### ISSUE TITLE
ADMIN_PASS not being set by AWS secret

### ISSUE THREAD

**Issue by mikeoleary** [F5Networks/f5-bigip-runtime-init#3]
# Description
I am trying to use f5-bigip-runtime-init to set my ADMIN_PASS from a AWS secret. It appears that f5-bigip-runtime-init finishes successfully but my admin password is not updated to the value of my AWS secret.

I am not sure what to expect when I log into the device, but after I change the password via command line and log into the GUI, I still see the Setup Utility. I need to click through the license page and resource provisioning, where only LTM is checked by default (despite my configuring ASM in the yaml file). Is the runtime-init configuration actually working? Interestingly:
- there are no mentions of failures in the log file, and the log file finishes by seeing TEEM report of success.
- the Automation Toolchain RPMs are already installed when I log in after deployment. So it is at least partly working.

What am I doing wrong?

# Documentation questions
In [this example](https://github.com/F5Networks/f5-bigip-runtime-init#aws-terraform-snippet) of setting a password via AWS secret manager, you refer to a Terraform secret version (ie, the secret version and not the secret itself). I have found that I need to pass the ARN of the AWS secret (ie, not the "name" of the secret but the "id" or the "arn") and this must be the secret, not the secret version. Does that make sense? I think the documentation could be clearer here, both that the ARN needs to be provided in the YAML file and that it's the secret, not the secret version.

# Troubleshooting done
I'm using silly log levels and cannot find any failures and I believe the following:
- I am using a string that I know is okay to use as a BIG-IP admin password. I can change the password to this value via TMSH, and I can also have this password change automated by using a python script to pull it from AWS Secrets Manager and setting this via user_data. But I would prefer to put as much onboarding in bigip-runtime-init as possible.
- I can see that BIG-IP is making an API call to retrieve secret value from AWS. CloudTrail shows that the Ec2 instance retrieves the value of the secret, and this API call appears to be successful. So I believe that fetching the value of the secret is not the problem.
- The IAM role that is attached to the EC2 instance only allows the action secretsmanager:GetSecretValue on the resource required. This is enough for me to use other methods (AWS cmd line tool and python script) to get the secret value, and I can see via CloudTrail that when using runtime-init, the API calls to get the secret appear successful. 
- /var/log/cloud/bigIpRuntimeInit.log shows a failure to find MAC address for interfaces 1.1 and 1.2 - is this a concern?

### My IAM role policy document 
```
data "aws_iam_policy_document" "bigip_policy" {
  version = "2012-10-17"
  statement {
    actions = [
      "secretsmanager:GetSecretValue"
    ]

    resources = [
      aws_secretsmanager_secret.bigip_password.arn
    ]
  }
}
```

### My YAML file
````yaml
---
runtime_parameters:
  - name: HOST_NAME
    type: metadata
    metadataProvider:
        environment: aws
        type: compute
        field: hostname
  - name: ADMIN_PASS
    type: secret
    secretProvider:
      environment: aws
      type: SecretsManager
      version: AWSCURRENT
      secretId: ${secret_id}
pre_onboard_enabled:
  - name: provision_rest
    type: inline
    commands:
      - /usr/bin/setdb provision.extramb 500
      - /usr/bin/setdb restjavad.useextramb true
  - name: provision_modules
    type: inline
    commands:
      - echo 'sys provision asm { level nominal }' >> bigip_base.conf
extension_packages:
  install_operations:
    - extensionType: do
      extensionVersion: 1.15.0
    - extensionType: as3
      extensionVersion: 3.20.0
    - extensionType: ts
      extensionVersion: 1.14.0
    - extensionType: cf
      extensionVersion: 1.5.0


    - extensionType: ilx
      extensionUrl: https://github.com/F5Networks/f5-appsvcs-templates/releases/download/v1.3.0/f5-appsvcs-templates-1.3.0-1.noarch.rpm
      extensionVersion: 1.3.0
      extensionVerificationEndpoint: /mgmt/shared/fast/info
extension_services:
  service_operations: []
````

**Comment by mikeoleary** [comment]
Hi,
Anyone able to help with this? I can't seem to get around this on my own. I've redeployed a few times, and from my logs it looks like the secret is successfully pulled from AWS secrets manager, but the password does not work when I try to log into BIG-IP. I see the log with ````"runtime_params":{"secrets":1,"metadata":1},"vaults":{"aws":1,"azure":0,"gcp":0}```` so it looks to me like the secret is successfully pulled from the secrets manager.

Here is the last line in my log file. A SUCCESS message appears to be sent in f5-teem report.
````
2020-11-13T21:56:29.824Z [19676]: debug: F5 TEEM Payload: {"platform":{"platform":"BIG-IP","platformVersion":"15.1.1","platformId":"Z100","system":{"cpuCount":4,"memory":16047,"diskSize":77824},"nicCount":3,"regKey":"[removed]","deployment":{"cloud":"aws"},"modules":{"ltm":"nominal"},"packages":{"f5-service-discovery-1.2.9-2.noarch":"1.2.9","f5-declarative-onboarding-1.15.0-3.noarch":"1.15.0","f5-cloud-failover-1.5.0-0.noarch":"1.5.0","f5-appsvcs-3.20.0-3.noarch":"3.20.0","f5-telemetry-1.14.0-2.noarch":"1.14.0","f5-appsvcs-templates-1.3.0-1.noarch":"1.3.0"},"environment":{"pythonVersion":"Python 2.7.5","pythonVersionDetailed":"2.7.5 (default, Oct  8 2020, 01:41:11) \n[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]","nodeVersion":"v6.9.1","libraries":{"ssh":"OpenSSH_7.4p1, OpenSSL 1.0.2s-fips  28 May 2019"}}},"product":{"version":"1.0.0","locale":"en-US","installDate":"2020-11-13T21:56:29.823Z","installationId":"[removed]","installedComponents":{"commander":"^4.1.0","winston":"^3.3.3","get-user-locale":"^1.4.0","uuid":"^8.2.0","@f5devcentral/f5-teem":"^1.4.6","js-yaml":"^3.13.1","mustache":"^4.0.0","request":"^2.88.0","aws-sdk":"^2.610.0","lodash.where":"^3.1.0"}},"operation":{"clientRequestId":"[removed]","rawCommand":"f5-runtime-init -c /config/cloud/cloud_config.yaml","pre_onboard_enabled":{"commands":2},"runtime_params":{"secrets":1,"metadata":1},"vaults":{"aws":1,"azure":0,"gcp":0},"userAgent":"f5-bigip-runtime-init/1.0.0","extension_packages":{"do":"1.15.0","as3":"3.20.0","ts":"1.14.0","cf":"1.5.0","ilx":"1.3.0"},"extension_services":{},"post_onboard_enabled":{"commands":0,"postHooks":0},"result":"SUCCESS","resultSummary":"Configuration Successful","startTime":"2020-11-13T21:55:56.897Z","endTime":"2020-11-13T21:56:28.141Z"}}
````

**Comment by mikeoleary** [comment]
I have finally worked this out and it's my fault. I thought my cloud_config.yaml file, outlined above, was sufficient to set the password on BIG-IP, but I had not included a reference to a DO declaration that would set my password to {{ADMIN_PASS}}. I mistakenly thought that simply pulling the aws secret was enough to set the password, but I see I need a DO declaration to accompany this. I've created a DO declaration, referenced the url in my service_operations section of my yaml file, and this now works. I re-read the documentation many times before it finally clicked for me, sorry! I'll close this issue now.
